File: ts/src/test/base/test.urlencode.ts
import assert from 'assert';
import ccxt from '../../../ccxt.js';

function testUrlencode () {

    const exchange = new ccxt.Exchange ({
        'id': 'sampleexchange',
    });

    // todo: add nulls
    // todo: add sort
    const dict1 = {
        'a': 1,
        'c': '+&',
    };
    // as key-order not preserved, expect mixed order
    const expected1 = 'a=1&c=%2B%26';
    const expected2 = 'c=%2B%26&a=1';
    const encoded = exchange.urlencode (dict1);
    assert (encoded === expected1 || encoded === expected2, 'testUrlencode: expected ' + expected1 + ' or ' + expected2 + ' but got ' + encoded);
}

export default testUrlencode;

File: julia/Ccxt/test/base/test.urlencode.jl
using Test
using Ccxt
function testUrlencode()

    exchange = get(ccxt, Symbol("Exchange"), nothing)(Dict{Symbol, Any}(
        Symbol("id") => "sampleexchange"
    ));
    dict1 = Dict{Symbol, Any}(
        Symbol("a") => 1,
        Symbol("c") => "+&"
    );
    expected1 = "a=1&c=%2B%26";
    expected2 = "c=%2B%26&a=1";
    encoded = urlencode(exchange, dict1);
    @test functions.ccxtruthy(@functions.ccxt_or(encoded == expected1, encoded == expected2))
end
